app/assets/javascripts/components/AgencySignupFields.jsx

```diff
@@ -0,0 +1,42 @@
+class AgencySignupFields extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { disableAgency: false }
+    this.emailChanged = this.emailChanged.bind(this);
+  }
+
+  emailChanged(e) {
+    this.forceUpdate();
+
+    if(e.target.value.indexOf('@hifinfo.org') > -1) {
+      this.setState({disableAgency: true});
+    } else {
+      this.setState({disableAgency: false});
+    }
+  }
```

Comment:
this can be condensed:

`let disableAgency  = e.target.value.indexOf('@hifinfo.org') > -1;

this.setState({disableAgency: disableAgency});
`